src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal class LocalUsingVarRewriter : BoundTreeRewriterWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
+    {
+        public static BoundNode Rewrite(BoundStatement statement)
+        {
+            var localUsingVarRewriter = new LocalUsingVarRewriter();
+            return localUsingVarRewriter.Visit(statement);
+        }
+
+        public override BoundNode VisitBlock(BoundBlock node)
+        {
+            ImmutableArray<BoundStatement> statements = (ImmutableArray<BoundStatement>)this.VisitList(node.Statements);
+            for (int i = 0; i < statements.Length; i++)
+            {
+                if (statements[i] is BoundLocalDeclaration localDeclaration)
+                {
+                    if (localDeclaration.LocalSymbol.IsUsing)
+                    {
+                        return LowerBoundLocalDeclarationUsingVar(node, statements);
+                    }
+                }
+                else if (statements[i] is BoundMultipleLocalDeclarations boundMultiple)
+                {
+                    if (!boundMultiple.LocalDeclarations.IsDefaultOrEmpty && boundMultiple.LocalDeclarations[0].LocalSymbol.IsUsing)
+                    {
+                        ArrayBuilder<BoundStatement> precedingStatements = ArrayBuilder<BoundStatement>.GetInstance(i);
+                        for (int j = 0; j < i; j++)
+                        {
+                            precedingStatements.Add(statements[j]);
+                        }
+                        var followingStatements = ImmutableArray.Create(statements, i + 1, statements.Length - i - 1);
+                        return LowerBoundMultipleLocalDeclarationUsingVar(boundMultiple, node.Locals, precedingStatements, followingStatements);
```

Comment:
>precedingStatements [](start = 102, length = 19)

Consider adding a comment to clarify that `precedingStatements` will be freed by `LowerBoundMultipleLocalDeclarationUsingVar`

Reply:
Consider freeing here, in the same method that allocated the builder.

---
In reply to: [205265116](https://github.com/dotnet/roslyn/pull/28587#discussion_r205265116) [](ancestors = 205265116)

Reply:
@cston I can't do that, I need to append to the end of preceding statements in the function it is passed into. Will create comment to clarify

Reply:
Perhaps:
```
var result = LowerBoundMultipleLocalDeclarationsUsingVar(..., precedingStatements, followingStatements);
precedingStatements.Free();
return result;
```

---
In reply to: [205279776](https://github.com/dotnet/roslyn/pull/28587#discussion_r205279776) [](ancestors = 205279776)